service/gcsutils/gcstools/netnsConfig.go

```diff
@@ -98,6 +98,13 @@ func netnsConfig() error {
 		return fmt.Errorf("netlink.LinkByName(%s) failed: %v", *ifStr, err)
 	}
 
+	// User requested non-default MTU size
+	if a.EncapOverhead != 0 {
+		if err = netlink.LinkSetMTU(link, int(a.EncapOverhead)); err != nil {
```

Comment:
I would have not expected from the name `EncapOverhead` that it actually contains the MTU. If `EncapOverhead` encodes the overhead required for encapsulation, I would have assumed this code to read the current MTU, deduct the overhead, and then set the MTU to the reduced value.